doc/source/dev_guide/plugins.rst

```diff
@@ -1,34 +1,207 @@
+needing to explicitly import your package. This has the added
+benefit of organizing your YAML configuration files and any custom python code
+into a single python package. How to structure a package in this way is
+described below.
+
+An example project showing the usage of these entry points is available at
+`this github repository <https://github.com/pytroll/satpy-composites-plugin-example>`_
+where a custom compositor is created. This repository also includes common
+configuration files and tools for writing clean code and automatically testing
+your python code.
+
+Plugin package structure
+========================
+
+The below sections will use the example package name ``satpy-myplugin``. This
+is only an example and naming a plugin package with a ``satpy-`` prefix is not
+required.
+
+A plugin package should consist of three main parts:
+
+1. ``pyproject.toml`` or ``setup.py``: These files define the metadata and
+   entry points for your package. Only one of them is needed. With only a few
+   exceptions it is recommended to use a ``pyproject.toml`` as this is the new
+   and future way Python package configuration will be supported by the ``pip``
+   package manager. See below for examples of the contents of this file.
+2. ``mypkg/etc/``: A directory of Satpy-compatible component YAML files. These
+   YAML files should be in ``readers/``, ``composites/``, ``enhancements/``,
+   and ``writers/`` directories. These YAML files must follow the Satpy naming
+   conventions for each component. For example, composites and enhancements
+   allow for sensor-specific configuration files. Satpy will collect all
+   available YAML files from all installed plugins and merge them with those
+   builtin to Satpy. The Satpy builtins will be used as a "base" configuration
+   with all external YAML files applied after.
+3. ``mypkg/``: The python package with any custom python code. This code should
+   be based on Satpy's base classes for each component or use utilities
+   available from Satpy whenever possible.
+
+   * readers: :class:`~satpy.readers.yaml_reader.FileYAMLReader` for any
+     reader subclasses and
+     :class:`~satpy.readers.file_handlers.BaseFileHandler` for any custom file
+     handlers. See :doc:`custom_reader` for more information.
+   * composites and modifiers: :class:`~satpy.composites.CompositeBase` for
+     any generic compositor and :class:`~satpy.composites.GenericCompositor`
+     for any composite that represents an image (RGB, L, etc). For modifiers,
+     use :class:`~satpy.modifiers.ModifierBase`.
+   * enhancements: Although not required, consider using
+     :func:`satpy.enhancements.apply_enhancement`.
+   * writers: :class:`~satpy.writers.Writer`
+
+   Lastly, this directory should be structured like a standard python package.
+   This primarily means a ``mypkg/__init__.py`` file should exist.
+
+pyproject.toml
+--------------
+
+A ``pyproject.toml`` file can be used to define the metadata and configuration
```

Comment:
Maybe we should say that's the recommended method? like `We recommend using a pyproject.toml file to define the metadata and configuration...`